### ISSUE TITLE
cpu.corespersocket don't work

### ISSUE THREAD

**Issue by edikevich** [apache/cloudstack#5695]
##### ISSUE TYPE
 * Bug Report

##### CLOUDSTACK VERSION
~~~
4.16.0
~~~

##### CONFIGURATION
XCP-NG 8.2
Advanced Zone

##### OS / ENVIRONMENT
Oracle Linux 8.4 - Management server, MYSQL 8.0.26
XCP-NG 8.2 - hypervisor

##### SUMMARY
cpu.corespersocket don't work

##### STEPS TO REPRODUCE
~~~
Create VM. Stop VM. Go to Setting. Add cpu.corespersocket = 2 (VM have 2 cores or more)
~~~
![image](https://user-images.githubusercontent.com/42668237/142237236-2432acb9-3b4a-4d75-b570-fe2a409fb236.png)

##### EXPECTED RESULTS
~~~
In XOCE, XCP-NG Center or where ever we see 1 sockets with 2 core per socket
~~~

##### ACTUAL RESULTS
~~~
In XOCE, XCP-NG Center or where ever we see 2 sockets with 1 core per socket
~~~

**Comment by nvazquez** [comment]
Hi @edikevich can you try the following steps?

- Stop VM
- Go to Settings -> Edit the existing platform setting and add the following `;cores-per-socket:2`
- Start VM
<img width="961" alt="Screen Shot 2022-01-21 at 16 04 16" src="https://user-images.githubusercontent.com/5295080/150585303-b40663ff-c564-48f2-8bab-f5b6760faa15.png">

After that I could verify the param was set, with the following command:
````
# xe vm-param-list uuid=70c700c5-14f3-c6b5-91a4-0175a41a61b5 | grep cores
                              platform (MRW): timeoffset: 0; device-model: qemu-upstream-compat; vga: std; videoram: 8; apic: true; viridian: false; pae: true; cores-per-socket: 2; acpi: 1; hpet: true; secureboot: false; nx: true
````

**Comment by edikevich** [comment]
Hi @nvazquez 
Yes, this is works!

**Comment by nvazquez** [comment]
Thanks for verifying @edikevich
@DaanHoogland @sureshanaparti do you think applying the setting ‘cpu.corespersocket’ should behave as the workaround in the ‘plarform’ setting?

**Comment by DaanHoogland** [comment]
yes , @nvazquez . But would that mean that the value is set twice? once as separate setting and once as a part of `platform`settings? I don't think that is desirable. How would the user control this?
intuitively, i'd expect `platform` to be read-only and always composed of separate parameters. Or alternatively to be synced back and forth to the individual parameters.

**Comment by sureshanaparti** [comment]
@nvazquez When the `platform` setting exists,  `cpu.corespersocket` setting is ignored. Will this work if we set _cores-per-socket_ from the  `cpu.corespersocket` setting in the platform irrespective of `platform` setting.

https://github.com/apache/cloudstack/blob/97d6cd50a05e06c532ae970d3a2ab737664f8b5f/plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java#L1917-L1922

**Comment by sureshanaparti** [comment]
Fixed in #5892
